backend/dataall/core/environment/services/environment_service.py

```diff
@@ -814,32 +814,49 @@ def delete_environment(session, uri, environment):
             .filter(EnvironmentGroup.environmentUri == uri)
             .all()
         )
-        for group in env_groups:
-
-            session.delete(group)
-
-            ResourcePolicy.delete_resource_policy(
-                session=session,
-                resource_uri=uri,
-                group=group.groupUri,
-            )
-
         env_roles = (
             session.query(ConsumptionRole)
             .filter(ConsumptionRole.environmentUri == uri)
             .all()
         )
+
+        env_resources = 0
+        for group in env_groups:
+            env_resources += EnvironmentResourceManager.count_group_resources(
+                session,
+                environment,
+                group.groupUri
+            )
         for role in env_roles:
-            session.delete(role)
+            env_resources += EnvironmentResourceManager.count_consumption_role_resources(
+                session,
+                environment,
+                role.consumptionRoleUri
+            )
 
-        KeyValueTag.delete_key_value_tags(
-            session, environment.environmentUri, 'environment'
-        )
+        if env_resources > 0:
+            raise Exception
```

Comment:
We should add a message for the exception.

Reply:
I moved the exception handling to the EnvironmentService - previously handling in the env resolver